velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -713,6 +749,33 @@ MapVectorPtr VectorFuzzer::fuzzMap(
       values);
 }
 
+FlatMapVectorPtr VectorFuzzer::fuzzFlatMap(
+    const TypePtr& keyType,
+    const TypePtr& valueType,
+    vector_size_t size) {
+  // generate a vector of keys and extract the distinct ones
+  auto allKeys = fuzzFlatNotNull(keyType, getElementsVectorLength(opts_, size));
+  VectorPtr distinctKeys = VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+      extractDistinctKeys, keyType->kind(), allKeys, pool_);
+
+  // inMap buffer and mapValues vector should be the same size
+  std::vector<VectorPtr> mapValues(distinctKeys->size());
+  std::vector<BufferPtr> inMaps(distinctKeys->size());
```

Comment:
let's also toss a coin and leave a few inMap buffers nullptr (this should mean all keys pairs are in the map)